### PR TITLE
Remove Spurious Blank Lines in PRT File Reports

### PR DESCRIPTION
**Pull request by bska**
The [`StreamLog::addMessageUnconditionally()`](https://github.com/OPM/opm-common/blob/cfa454e07587ea8ba6cd170760fad8da0e3b1ec2/opm/common/OpmLog/StreamLog.hpp#L38) member function will [end each message with a newline](https://github.com/OPM/opm-common/blob/cfa454e07587ea8ba6cd170760fad8da0e3b1ec2/src/opm/common/OpmLog/StreamLog.cpp#L56) (`std::endl`) so we should not add such newlines ourselves.  The extra newline characters produce spurious blank lines in the report sheets, e.g., for the "PRODUCTION REPORT".

This PR removes the last newline character from each report request, thus deferring that responsibility to `OpmLog::note()` instead.  Doing so, however, means we have take a little more care with the first line of each report lest we create report sheets which are smushed together.

While here, also pull substep checking out of the the output module's `outputProdLog()`, `outputInjLog()`, and `outputCumLog()` member functions.  This parameter was only used in the same way in internal conditions in each member function and we can enforce that check on the outside without losing expressivity.

---

#### Current Master ####
```
Starting time step 2, stepsize 4 days, at day 4/8, date = 10-Nov-1997
======================================================= PRODUCTION REPORT =======================================================
:  WELL  :  LOCATION :CTRL:    OIL    :   WATER   :    GAS    :   FLUID   :   WATER   : GAS/OIL  :  WAT/GAS   : BHP OR : THP OR :
:  NAME  :  (I,J,K)  :MODE:    RATE   :   RATE    :    RATE   :  RES.VOL. :    CUT    :  RATIO   :   RATIO    : CON.PR.: BLK.PR.:
:        :           :    :  SCM/DAY  :  SCM/DAY  :  SCM/DAY  :  RCM/DAY  :  SCM/SCM  :  SCM/SCM :  SCM/SCM   :  BARSA :  BARSA :
=================================================================================================================================

:   FIELD:           :    :     4371.2:        4.5:   474638.8:     5733.4:      0.001:    108.58:      0.0000:        :        :
:--------:-----------:----:-----------:-----------:-----------:-----------:-----------:----------:------------:--------:--------:

:  MANI-C:           :    :       -0.0:       -0.0:       -0.0:       -0.0:      0.000:      0.00:      0.0000:        :        :
:--------:-----------:----:-----------:-----------:-----------:-----------:-----------:----------:------------:--------:--------:

[...]
:--------:-----------:----:-----------:-----------:-----------:-----------:-----------:----------:------------:--------:--------:

:D2-DUMMY:           :    :       -0.0:       -0.0:       -0.0:       -0.0:      0.000:      0.00:      0.0000:        :        :
:--------:-----------:----:-----------:-----------:-----------:-----------:-----------:----------:------------:--------:--------:                                                             


=================================================== INJECTION REPORT ========================================
:  WELL  :  LOCATION : CTRL : CTRL : CTRL :    OIL    :   WATER   :    GAS    :   FLUID   : BHP OR : THP OR :
:  NAME  :  (I,J,K)  : MODE : MODE : MODE :    RATE   :   RATE    :    RATE   :  RES.VOL. : CON.PR.: BLK.PR.:
:        :           : OIL  : WAT  : GAS  :  SCM/DAY  :  SCM/DAY  :  SCM/DAY  :  RCM/DAY  :  BARSA :  BARSA :
==============================================================================================================

:   FIELD:           :      :      :      :        0.0:        0.0:        0.0:        0.0:        :        :
:--------:-----------:------:------:------:------------:----------:-----------:-----------:--------:--------:

:  MANI-C:           :      :      :      :        0.0:        0.0:        0.0:        0.0:        :        :
:--------:-----------:------:------:------:------------:----------:-----------:-----------:--------:--------:

[...]
:--------:-----------:------:------:------:------------:----------:-----------:-----------:--------:--------:

:D2-DUMMY:           :      :      :      :        0.0:        0.0:        0.0:        0.0:        :        :
:--------:-----------:------:------:------:------------:----------:-----------:-----------:--------:--------:

=================================================== CUMULATIVE PRODUCTION/INJECTION REPORT =========================================
:  WELL  :  LOCATION :  WELL  :CTRL:    OIL    :   WATER   :    GAS    :   Prod    :    OIL    :   WATER   :    GAS    :   INJ     :
:  NAME  :  (I,J,K)  :  TYPE  :MODE:    PROD   :   PROD    :    PROD   :  RES.VOL. :    INJ    :   INJ     :    INJ    :  RES.VOL. :
:        :           :        :    :    MSCM   :   MSCM    :    MMSCM  :   MRCM    :    MSCM   :   MSCM    :    MMSCM  :   MRCM    :
====================================================================================================================================

:   FIELD:           :        :    :       35.0:        0.0:     3797.9:       45.9:        0.0:        0.0:        0.0:        0.0:
:--------:-----------:--------:----:------------:----------:-----------:-----------:------------:----------:-----------:-----------:

:  MANI-C:           :        :    :        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:
:--------:-----------:--------:----:------------:----------:-----------:-----------:------------:----------:-----------:-----------:

[...]
:--------:-----------:--------:----:------------:----------:-----------:-----------:------------:----------:-----------:-----------:

:D2-DUMMY:           :        :    :        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:
:--------:-----------:--------:----:------------:----------:-----------:-----------:------------:----------:-----------:-----------:

 Newton its= 2, linearizations= 3 (0.0sec), linear its=  2 (0.1sec)
```

#### This PR ####
```
Starting time step 2, stepsize 4 days, at day 4/8, date = 10-Nov-1997

======================================================= PRODUCTION REPORT =======================================================
:  WELL  :  LOCATION :CTRL:    OIL    :   WATER   :    GAS    :   FLUID   :   WATER   : GAS/OIL  :  WAT/GAS   : BHP OR : THP OR :
:  NAME  :  (I,J,K)  :MODE:    RATE   :   RATE    :    RATE   :  RES.VOL. :    CUT    :  RATIO   :   RATIO    : CON.PR.: BLK.PR.:
:        :           :    :  SCM/DAY  :  SCM/DAY  :  SCM/DAY  :  RCM/DAY  :  SCM/SCM  :  SCM/SCM :  SCM/SCM   :  BARSA :  BARSA :
=================================================================================================================================
:   FIELD:           :    :     4371.2:        4.5:   474638.8:     5733.4:      0.001:    108.58:      0.0000:        :        :
:--------:-----------:----:-----------:-----------:-----------:-----------:-----------:----------:------------:--------:--------:
:  MANI-C:           :    :       -0.0:       -0.0:       -0.0:       -0.0:      0.000:      0.00:      0.0000:        :        :
:--------:-----------:----:-----------:-----------:-----------:-----------:-----------:----------:------------:--------:--------:
[...]
:--------:-----------:----:-----------:-----------:-----------:-----------:-----------:----------:------------:--------:--------:
:D2-DUMMY:           :    :       -0.0:       -0.0:       -0.0:       -0.0:      0.000:      0.00:      0.0000:        :        :
:--------:-----------:----:-----------:-----------:-----------:-----------:-----------:----------:------------:--------:--------:

=================================================== INJECTION REPORT ========================================
:  WELL  :  LOCATION : CTRL : CTRL : CTRL :    OIL    :   WATER   :    GAS    :   FLUID   : BHP OR : THP OR :
:  NAME  :  (I,J,K)  : MODE : MODE : MODE :    RATE   :   RATE    :    RATE   :  RES.VOL. : CON.PR.: BLK.PR.:
:        :           : OIL  : WAT  : GAS  :  SCM/DAY  :  SCM/DAY  :  SCM/DAY  :  RCM/DAY  :  BARSA :  BARSA :
==============================================================================================================
:   FIELD:           :      :      :      :        0.0:        0.0:        0.0:        0.0:        :        :
:--------:-----------:------:------:------:-----------:-----------:-----------:-----------:--------:--------:
:  MANI-C:           :      :      :      :        0.0:        0.0:        0.0:        0.0:        :        :
:--------:-----------:------:------:------:-----------:-----------:-----------:-----------:--------:--------:
[...]
:--------:-----------:------:------:------:-----------:-----------:-----------:-----------:--------:--------:
:D2-DUMMY:           :      :      :      :        0.0:        0.0:        0.0:        0.0:        :        :
:--------:-----------:------:------:------:-----------:-----------:-----------:-----------:--------:--------:

=================================================== CUMULATIVE PRODUCTION/INJECTION REPORT =========================================
:  WELL  :  LOCATION :  WELL  :CTRL:    OIL    :   WATER   :    GAS    :   Prod    :    OIL    :   WATER   :    GAS    :   INJ     :
:  NAME  :  (I,J,K)  :  TYPE  :MODE:    PROD   :   PROD    :    PROD   :  RES.VOL. :    INJ    :   INJ     :    INJ    :  RES.VOL. :
:        :           :        :    :    MSCM   :   MSCM    :    MMSCM  :   MRCM    :    MSCM   :   MSCM    :    MMSCM  :   MRCM    :
====================================================================================================================================
:   FIELD:           :        :    :       35.0:        0.0:     3797.9:       45.9:        0.0:        0.0:        0.0:        0.0:
:--------:-----------:--------:----:-----------:-----------:-----------:-----------:------------:----------:-----------:-----------:
:  MANI-C:           :        :    :        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:
:--------:-----------:--------:----:-----------:-----------:-----------:-----------:------------:----------:-----------:-----------:
[...]
:--------:-----------:--------:----:-----------:-----------:-----------:-----------:------------:----------:-----------:-----------:
:D2-DUMMY:           :        :    :        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:        0.0:
:--------:-----------:--------:----:-----------:-----------:-----------:-----------:------------:----------:-----------:-----------:
```

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Review by akva2 — APPROVED** [review]
This is fine. I think we can also nuke the other bool params but maybe I'm missing something.

File: ebos/eclwriter.hh [hunk 1]
```diff
@@ -311,11 +308,16 @@ public:
...
-        OPM_TIMEBLOCK(outputXXX);
-        eclOutputModule_->outputProdLog(reportStepNum, isSubStep, forceDisableProdOutput);
-        eclOutputModule_->outputInjLog(reportStepNum, isSubStep, forceDisableInjOutput);
-        eclOutputModule_->outputCumLog(reportStepNum, isSubStep, forceDisableCumOutput);
+        if (! isSubStep) {
+            OPM_TIMEBLOCK(outputProdInjLogs);
+
+            const bool forceDisableProdOutput = false;
+            const bool forceDisableInjOutput = false;
+            const bool forceDisableCumOutput = false;
```
[the remark this answers is not in the document]
Comment (bska) [reply ebos/eclwriter.hh:316]: > while at it, these bools seems pretty dumb as well?

I don't disagree.  They were introduced in PR #2073, **probably** as means to turning off the reports through a command line parameter, but that idea appears to have been rejected during the review and never fully removed.  I'd be okay with removing them now, with the option of reintroducing something similar if we want the ability.

**Comment by bska** [comment]
> This is fine. I think we can also nuke the other bool params but maybe I'm missing something.

I've pushed an update to remove those parameters now.  Heads up, however, I intend to make a few more compatibility related changes in this area.

**Comment by bska** [comment]
PR approved and build check is green.  I'll merge into master.

**Pull request merged**